apps/api/src/campaign/campaign.service.ts

```diff
@@ -553,12 +529,17 @@ export class CampaignService {
           },
           select: donationNotificationSelect,
         })
+
+        //if donation is switching to successful, increment the vault amount and send notification
         if (newDonationStatus === DonationStatus.succeeded) {
+          await this.vaultService.incrementVaultAmount(donation.targetVaultId, donation.amount, tx)
           this.notificationService.sendNotification('successfulDonation', {
             ...updatedDonation,
-            person: donation.person,
+            person: updatedDonation.person,
           })
         }
+
```

Comment:
should we add the `
 if (metadata?.isAnonymous != 'true') {
        await this.prisma.donation.update({
          where: { id: donation.id  ......
` logic for associating an email with a donation when updating